server/data/welcomeClient.ts

```diff
@@ -129,4 +131,26 @@ export default class WelcomeClient {
       path: '/prison/users/me/caseLoads',
     }) as Promise<UserCaseLoad[]>
   }
+
+  async getMatchingRecords(matchCriteria: PotentialMatchCriteria): Promise<PotentialMatch[]> | null {
+    logger.info(`welcomeApi: match-prisoners`)
+    try {
+      return (await this.restClient.post({
+        path: '/match-prisoners',
+        data: matchCriteria,
+      })) as Promise<PotentialMatch[]>
+    } catch (error) {
+      if (error.status >= 400 && error.status < 500) {
```

Comment:
we shouldn't need this as we can assume that there will always be an empty list if there are no matches

Reply:
ok